src/front/app/components/Challenges/ChallengeList.js

```diff
@@ -1,7 +1,47 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import Challenge from './Challenge';
+import TopChallenge from './TopChallenge';
 
-export default () => (
-    <div>
-        ChallengeList
-    </div>
-);
+let ChallengeList = ({ challenges, dispatch }) => {
+    if (!challenges.length) {
```

Comment:
Component data initialization logic should be moved out of the render function, as its responsibility is only to return a document fragment based on the state and properties received and processed earlier in the component life cycle